Yoast/YoastSEO.js
# Fix some eslint warnings in YoastSEO.js

## Summary

This PR can be summarized in the following changelog entry:

* [no-user-facing-changes] Fixes a number of `require-jsdoc` and `no-shadow` eslint warnings.
* Fixes a but where a Flesch reading ease score of exactly 90 would incorrectly trigger bad feedback.

## Relevant technical choices:

* Does not fix warnings in `app.js`and `appSpec.js` because they will be deprecated anyway #2045

## Test instructions

This PR can be tested by following these steps:

* Run `grunt check` and check if nothing fails.
* Run `yarn test` and check if nothing fails.
* Since this PR included some refactoring of the **Flesch Reading Ease assessment**, please check based on this information whether you can get all possible feedback strings:
  * **Red 3** Flesch Reading Ease: The copy scores 0 in the test, which is considered very difficult to read. Try to make shorter sentences, using less difficult words to improve readability.
  * **Red 3** Flesch Reading Ease: The copy scores 45 in the test, which is considered difficult to read. Try to make shorter sentences, using less difficult words to improve readability.
  * **Orange 6** Flesch Reading Ease: The copy scores 55 in the test, which is considered fairly difficult to read. Try to make shorter sentences to improve readability.
  * **Green 9** Flesch Reading Ease: The copy scores 63.9 in the test, which is considered ok to read. Good job!
  * **Green 9** Flesch Reading Ease: The copy scores X in the test, which is considered fairly easy to read. Good job!
  * **Green 9** Flesch Reading Ease: The copy scores X in the test, which is considered easy to read. Good job!
  * **Green 9** Flesch Reading Ease: The copy scores X in the test, which is considered very easy to read. Good job!
  * Check if the assessment title link works
  * Check if the assessment CTA link appears for all Orange and Red results and works.
  * Check if the language-specific configurations are still applied. For instance, try to get a Russian paper that scores between 40 and 50 and make sure to get an orange bullet (the same score for an English paper should result in a red bullet).


Fixes #1847 Fixes #1848
